src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AttendanceResourceIntTest.kt

```diff
@@ -640,8 +660,19 @@ class AttendanceResourceIntTest : IntegrationTestBase() {
     inner class DuplicateAttendance {
       @Test
       fun `duplicate attendance can be worked around by deleting one of them`() {
-        saveAttendance("SCH", courseSchedule, allocation)
-        val duplicate = saveAttendance("SCH", courseSchedule, allocation)
+        testData(repository) {
+          offenderBooking = offender(nomsId = "A1234AR") {
+            booking(agencyLocationId = "LEI") {
+              courseAllocation(courseActivity) {
+                payBand()
+                courseAttendance(courseSchedule)
+                courseAttendance(courseSchedule)
+              }
+            }
+          }.latestBooking()
+        }
+        allocation = offenderBooking.offenderProgramProfiles.first()
+        val duplicate = allocation.offenderCourseAttendances.first()
```

Comment:
This is what the DSL looks like in action.

Initially I tried to put the attendances as a child of a course schedule - this seemed like the most natural place to put them, attending people to a session. However this caused a circular reference that I struggled to resolve:
* an attendance links a course allocation and a course schedule
* if both allocation and schedule have the same parent (course activity) then neither is saved to the DB until the course activity is
* this means the attendance cannot have a reference to both allocation and schedule - only one of them can be its parent and the other has not been created yet

For this reason we save the course activity and schedule first, then we can add the allocation to the schedule.